### PR TITLE
Better diagnostics when Xwayland can't be found or executed

### PR DESCRIPTION
**Pull request by AlanGriffiths**
Better diagnostics when Xwayland can't be found or executed. (Fixes: #1604)

### CODE REVIEW

**Comment by AlanGriffiths** [comment]
> It would be far better for X apps to not work than to refuse to let them log in.

Really? Wouldn't it be better to know about a problem straight away instead of carrying with the effect that random apps do not work and having to guess which log file has a useful error message?

**Comment by wmww** [comment]
With Mir running in production, users wont necessarily see the text. What we *should* do is throw up an error message on-screen, but failing that IMO we should make do the best we can. Completely refusing to start Mir just because Xwayland is missing will lead to frustrated users, especially on systems where Mir is the only display server installed.

**Comment by Saviq** [comment]
`enable-x11` is off by default, right? In that case I'm with @AlanGriffiths, if it was explicitly enabled on our command line, then it makes sense to bail. Dependencies should ensure that XWayland is around, and if it's not, we're not really in a position to be debugging that.

If the session fails altogether, it's going to be a clear message that they need to look in the session logs for what's gone wrong.

The two asks that I have is maybe having a clearer error message and a specific exit code? Especially in the second case, where the `exec` fails, is there an error message from the OS we could forward. And a session wrapper could deal with restarting without `--enable-x11` if they see the exit code telling it that's what's failed.

File: src/server/frontend_xwayland/xwayland_server.cpp [hunk 1]
```diff
@@ -117,7 +117,7 @@ auto fork_xwayland_process(
 
     case 0:
         exec_xwayland(spawner, xwayland_path, wayland_pipe.client, x11_wm_pipe.server);
-        perror("Failed to start XWayland, should be unreachable");
+        mir::log_error("Cannot execute Xwayland: --xwayland-path %s", xwayland_path.c_str());
```
Comment (wmww) [inline src/server/frontend_xwayland/xwayland_server.cpp:120]: Is using `log_error()` in another process a good idea? Even if so, I still want the errno message to be printed. Alternative here: fcf5af1a6329d9d460cbc71360a66e64fcf8a2b5

**Pull request merged**